### PR TITLE
Ability popup updates

### LINKED ISSUE #2374: The ability pop up doesn't show all the characters properly for Pokémon with full nicknames
Title. I'm noticing that the ability pop up isn't showing a `'s` that is appended to a Pokémon's name completely InGame.

mGBA_20221004_001432266

While I'm not using an up-to-date branch, it's *almost* up-to-date. The only commits I'm missing are the last 2 in the upcoming branch, which do not affect the ability pop up in any way. It's a fix for the Berserk ability.

**EDIT**: This is only, explicitly happening when your Pokémon has a nickname that uses all the available characters. A single M less and it all shows up nicely InGame.

### PR DESCRIPTION
**Pull request by Jaizu**
## Description
Fixes #2374
Allows larger Pokémon/Nickname names
Small cleanup and documentation

## **Discord contact info**
Jaizu#0001

### CODE REVIEW

File: src/battle_interface.c [hunk 1]
```diff
@@ -2801,66 +2801,25 @@ static const struct SpritePalette sSpritePalette_AbilityPopUp =
...
-};
-
-static const union AnimCmd *const sSpriteAnimTable_AbilityPopUp1[] =
-{
-    sSpriteAnim_AbilityPopUp1
+	.affineMode = ST_OAM_AFFINE_OFF,
+	.objMode = ST_OAM_OBJ_NORMAL,
+	.shape = SPRITE_SHAPE(64x32),
+	.size = SPRITE_SIZE(64x32),
+	.priority = 0,
```
Comment (LOuroboros) [inline src/battle_interface.c:2808]: Tabs to spaces.

File: src/battle_interface.c [hunk 2]
```diff
@@ -2801,66 +2801,25 @@ static const struct SpritePalette sSpritePalette_AbilityPopUp =
...
-    .oam = &sOamData_AbilityPopUp,
-    .anims = sSpriteAnimTable_AbilityPopUp2,
+    .anims = gDummySpriteAnimTable,
     .images = NULL,
     .affineAnims = gDummySpriteAffineAnimTable,
     .callback = SpriteCb_AbilityPopUp
 };
 
-#define ABILITY_POP_UP_POS_X_DIFF 64
+#define ABILITY_POP_UP_POS_X_DIFF (64 - 7) //Hide second sprite underneath to gain proper letter spacing
```
Comment (LOuroboros) [inline src/battle_interface.c:2822]:
```suggestion
#define ABILITY_POP_UP_POS_X_DIFF (64 - 7) // Hide second sprite underneath to gain proper letter spacing
```

File: src/battle_interface.c [hunk 3]
```diff
@@ -2960,59 +2896,61 @@ static void ClearAbilityName(u8 spriteId1, u8 spriteId2)
...
 
-    lastChar = *(name - 1);
-    name[0] = CHAR_SGL_QUOTE_RIGHT; // apostraphe
-    name++;
+    if (*(textPtr - 1) == EOS)
+        textPtr--;
+
+    lastChar = *(textPtr - 1);
+
+    //Make the string say "[NAME]'s" instead of "[NAME]"
```
Comment (LOuroboros) [inline src/battle_interface.c:2927]:
```suggestion
    // Make the string say "[NAME]'s" instead of "[NAME]"
```

File: src/battle_interface.c [hunk 4]
```diff
@@ -3080,11 +3018,11 @@ static const u16 sOverwrittenPixelsTable[][2] =
     {PIXEL_COORDS_TO_OFFSET(8, 45), 8},
     {PIXEL_COORDS_TO_OFFSET(8, 46), 8},
     {PIXEL_COORDS_TO_OFFSET(8, 47), 8},
-    {PIXEL_COORDS_TO_OFFSET(8, 48), 8},
+    //{PIXEL_COORDS_TO_OFFSET(8, 48), 8},
```
Comment (LOuroboros) [inline src/battle_interface.c:3021]:
```suggestion
```

File: src/battle_interface.c [hunk 5]
```diff
@@ -3080,11 +3018,11 @@ static const u16 sOverwrittenPixelsTable[][2] =
...
     {PIXEL_COORDS_TO_OFFSET(16, 45), 8},
     {PIXEL_COORDS_TO_OFFSET(16, 46), 8},
     {PIXEL_COORDS_TO_OFFSET(16, 47), 8},
-    {PIXEL_COORDS_TO_OFFSET(16, 48), 8},
+    //{PIXEL_COORDS_TO_OFFSET(16, 48), 8},
```
Comment (LOuroboros) [inline src/battle_interface.c:3025]:
```suggestion
```

File: src/battle_interface.c [hunk 6]
```diff
@@ -3155,35 +3093,33 @@ void CreateAbilityPopUp(u8 battlerId, u32 ability, bool32 isDoubleBattle)
...
-                                coords[battlerPosition][1], 0);
-
-        gSprites[spriteId1].tOriginalX = coords[battlerPosition][0];
-        gSprites[spriteId2].tOriginalX = coords[battlerPosition][0] + ABILITY_POP_UP_POS_X_DIFF;
+		spriteId1 = CreateSprite(&sSpriteTemplate_AbilityPopUp,
+								coords[battlerPosition][0] - ABILITY_POP_UP_POS_X_SLIDE,
+								coords[battlerPosition][1], 0);
+		spriteId2 = CreateSprite(&sSpriteTemplate_AbilityPopUp,
+								coords[battlerPosition][0] - ABILITY_POP_UP_POS_X_SLIDE + ABILITY_POP_UP_POS_X_DIFF,
+								coords[battlerPosition][1], 1); //Appears below
```
Comment (LOuroboros) [inline src/battle_interface.c:3101]: Tabs to spaces.

File: src/battle_interface.c [hunk 7]
```diff
@@ -3155,35 +3093,33 @@ void CreateAbilityPopUp(u8 battlerId, u32 ability, bool32 isDoubleBattle)
...
-                                coords[battlerPosition][1], 0);
-
-        gSprites[spriteId1].tOriginalX = coords[battlerPosition][0];
-        gSprites[spriteId2].tOriginalX = coords[battlerPosition][0] + ABILITY_POP_UP_POS_X_DIFF;
+		spriteId1 = CreateSprite(&sSpriteTemplate_AbilityPopUp,
+								coords[battlerPosition][0] + ABILITY_POP_UP_POS_X_SLIDE,
+								coords[battlerPosition][1], 0);
+		spriteId2 = CreateSprite(&sSpriteTemplate_AbilityPopUp,
+								coords[battlerPosition][0] + ABILITY_POP_UP_POS_X_SLIDE + ABILITY_POP_UP_POS_X_DIFF,
+								coords[battlerPosition][1], 1); //Appears below
```
Comment (LOuroboros) [inline src/battle_interface.c:3113]: Tabs to spaces.

File: src/battle_interface.c [hunk 8]
```diff
@@ -3155,35 +3093,33 @@ void CreateAbilityPopUp(u8 battlerId, u32 ability, bool32 isDoubleBattle)
...
 
         gSprites[spriteId1].tRightToLeft = FALSE;
         gSprites[spriteId2].tRightToLeft = FALSE;
     }
 
+	gSprites[spriteId1].tOriginalX = coords[battlerPosition][0];
+	gSprites[spriteId2].tOriginalX = coords[battlerPosition][0] + ABILITY_POP_UP_POS_X_DIFF;
+	gSprites[spriteId2].oam.tileNum += (8 * 4); //Second half of pop up
+
```
Comment (LOuroboros) [inline src/battle_interface.c:3122]: Tabs to spaces.

**Review by LOuroboros — CHANGES_REQUESTED** [review]

**Comment by Jaizu** [comment]
Applied all the feedback.

**Review by AsparagusEduardo — APPROVED** [review]

**Pull request merged**